assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useUpdateEffect } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import {
+	MODULES_ANALYTICS_4,
+	PROPERTY_CREATE,
+	WEBDATASTREAM_CREATE,
+} from '../../../analytics-4/datastore/constants';
+import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
+import { Select, Option } from '../../../../material-components';
+import { GA4ActivateSwitch } from '../common';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../analytics-4/components/common';
+import ProgressBar from '../../../../components/ProgressBar';
+const { useSelect, useDispatch } = Data;
+
+export default function GA4SettingsControls() {
+	const [ matchedProperty, setMatchedProperty ] = useState();
+	const [ matchedWebDataStream, setMatchedWebDataStream ] = useState();
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// This select is needed to check whether the AdminAPI works or not.
+	useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+	);
+
+	const isAdminAPIWorking = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isAdminAPIWorking()
+	);
+
+	const enableGA4 = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'enableGA4' )
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const {
+		matchAccountProperty,
+		matchWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	} = useDispatch( MODULES_ANALYTICS_4 );
+
+	useUpdateEffect( () => {
+		const matchGA4Information = async () => {
+			const matchingProperty = await matchAccountProperty( accountID );
+
+			setMatchedProperty( matchingProperty );
+			if ( matchingProperty?._id ) {
+				const matchingWebDataStream = await matchWebDataStream(
+					matchingProperty._id
+				);
+				setMatchedWebDataStream( matchingWebDataStream );
+			}
+		};
+
+		if ( isAdminAPIWorking ) {
+			matchGA4Information();
+		}
+	}, [
+		accountID,
+		matchAccountProperty,
+		matchWebDataStream,
+		isAdminAPIWorking,
+	] );
+
+	const onActivate = useCallback( () => {
+		setPropertyID( matchedProperty?._id || PROPERTY_CREATE );
+		setWebDataStreamID( matchedWebDataStream?._id || WEBDATASTREAM_CREATE );
+		setMeasurementID( matchedWebDataStream?.measurementId || '' ); // eslint-disable-line sitekit/acronym-case
+	}, [
+		matchedProperty,
+		matchedWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	] );
+
+	if ( isAdminAPIWorking === undefined ) {
+		return <ProgressBar height={ 158 } small />;
```

Comment:
Where do you get the `158` from? In my testing:
* If `enableGA4`, that entire area is `184` on desktop and `168` on smaller viewports.
* If `! enableGA4`, that area is `152` on desktop and `128` on smaller viewports.
* In all cases, the bottom margin is `16px`.

What if we set this to `200` (`184 + 16`) if `enableGA4` and to `168` (`152 + 16`) if not? It still doesn't get _all_ cases, but at least I would think there is no layout shift on desktop? This is a bit of a nit-picking exercise though, so maybe we should just let it be for now 🤔

Reply:
`158` is the height of the `GA4SettingsControls` component when `! enableGA4` in my browser. If this is critical, let's create a follow-up issue for this and leave `158` for now. What do you think?

Reply:
@felixarntz I think this is the last important thing to finalize here and then this is good to go on my end.

Reply:
I just tested this again and pushed another commit https://github.com/google/site-kit-wp/pull/4229/commits/dfe5fd63c6feb38b950fd902a13d5577513a002f that fixes it for me (at least partially, i.e. on desktop viewports).

It now uses `180` and `212` for the heights, because of the following:
* Height of the actual UI for the disabled and enabled state on desktop are `168` and `200` respectively, per what I've described above.
* There's an additional `16px` margin from the container in the actual UI, so this would lead to `184` and `216` respectively.
* The `ProgressBar` itself is `4px` tall, so whatever height you pass to it will actually not be entirely correctly reflected, the overall height will be `4px` taller because the height passed as prop is only used for the margins. This is a problem in the `ProgressBar` component we should probably fix, but not worth doing here. Anyway, because of those `4px` we need to subtract them, leading to `180` and `212`.
* Of course this still doesn't account for smaller viewports. We can't properly fix that though because the `ProgressBar` component doesn't support multiple heights per viewport - another thing to eventually fix. Smaller viewports are much less commonly used for WordPress and Site Kit though, so we should focus on desktop for now.

@eugene-manuilov @aaemnnosttv Let me know if this sounds good to you for now.

Reply:
The layout shift is gone for me now as well 🎉